### ISSUE TITLE
Spike: geolocation tracking with a view to personalisation for local resources

### ISSUE THREAD

**Issue by Cleop** [LDMW/cms#146]
Research into geolocation tracking options available (wagtail/google analytics) in order to update resources according to a user's location if they are localised resources.

Personalised borough cover image?

**Comment by samhstn** [comment]
If using [`html5 Geolocation`](https://www.w3schools.com/html/html5_geolocation.asp) this would give us the latitude and longitude of the user.

We could then use something like [this api](http://www.latlong.net/Show-Latitude-Longitude.html) to convert that to a postcode.

We could have a picture stored by the admins with a corresponding first part of a postcode.
Each resource could also have a postcode field.

If we have a postcode match, should we put the resources to the top?

We could instead put the non matching postcodes to the bottom.

@Danwhy, @simonlab what do you think of going about it this way?

**Comment by samhstn** [comment]
@ellemindwave @sarahwelton How would you like us to request the users location?
Would it more likely be a button or a checkbox or should we ask for it on page load (page load would be just showing the popup which appears when clicking the `try it` button [here](https://www.w3schools.com/html/tryit.asp?filename=tryhtml5_geolocation))

**Comment by samhstn** [comment]
Initial spike was around 2h (ended up going for the google maps api).

I'll estimate the implementation of this to be just over a day.

**Comment by samhstn** [comment]
- [x] Set up the sending of the client location to the server
- [x] Server route will ping the google maps api and then respond with that users postcode to be stored in a cookie on the browser
- [x] Every subsequent request will also send that cookie data in the header which the server can use send different data back with

**Comment by reddog** [comment]
@shouston3 What are you using to determine the client's location?  There is the location information from the browser which is particularly useful on a mobile device (provided it's turned on).  What will be used (if anything?) from a non-mobile device or where browser location is turned off?

**Comment by samhstn** [comment]
I was planning on using [html5 geolocation](https://www.w3schools.com/html/html5_geolocation.asp) for all devices (at least as an initial proof of concept) and nothing when turned off.

@reddog What method would be best to go for? Could you link to a reference for the in browser location information you are referring to?

**Comment by reddog** [comment]
@shouston3 That sounds fine, providing it's available.  Just wanted to check on what would be used if it wasn't available (ie not a mobile or user has chosen not to share their location).

Would there be a default image (and message?).  Or a random choice, maybe?

Would this be the same if out of area (ie accessed outside of London, maybe even outside the UK!)

**Comment by samhstn** [comment]
I was planning that, as the proof of concept, we wouldn't customise the page for the user's location if they don't have it enabled or there isn't a postcode match and would in this case show the default image.

We could have a random image choice if you'd like (we can open a new issue for that).

**Comment by reddog** [comment]
@shouston3 I think that all sounds good.  I'll speak to the team about opening another issue about the random selection.

**Comment by samhstn** [comment]
Geolocation will be working once #158 is merged in - with different background images showing depending on if the user is in North, East, South or West London and the default image if the user doesn't tell us the location or the background image hasn't yet been configured. We can review geolocation requirements tomorrow, not continuing on this feature until then.

**Comment by samhstn** [comment]
@Cleop this spike has been spiked (and pr'ed here: #158). From the discussion yesterday we are still deciding what we want to do with this feature. Could we open a new issue for discussion on how to proceed from here and we can look at closing off this milestone

**Comment by samhstn** [comment]
It looks like this has already been moved into first steps: #191 and next steps: #184. @Cleop can we close this

**Comment by Cleop** [comment]
Closing as now linked to issues #191 and #184.
